src/plonk/lookup/prover.rs

```diff
@@ -576,24 +608,29 @@ type ExpressionPair<F> = (Polynomial<F, LagrangeCoeff>, Polynomial<F, LagrangeCo
 ///   that has the corresponding value in S'.
 /// This method returns (A', S') if no errors are encountered.
 fn permute_expression_pair<C: CurveAffine>(
+    pk: &ProvingKey<C>,
+    params: &Params<C>,
     domain: &EvaluationDomain<C::Scalar>,
     input_expression: &Polynomial<C::Scalar, LagrangeCoeff>,
     table_expression: &Polynomial<C::Scalar, LagrangeCoeff>,
 ) -> Result<ExpressionPair<C::Scalar>, Error> {
-    let mut permuted_input_expression = input_expression.clone();
+    let blinding_factors = pk.vk.cs.blinding_factors();
+    let mut permuted_input_expression: Vec<C::Scalar> = input_expression.to_vec();
+    permuted_input_expression.truncate(params.n as usize - (blinding_factors + 1));
```

Comment:
We use `params.n as usize - (blinding_factors + 1)` three times; consider defining a local variable for it.